bc_obps/service/document_service.py

```diff
@@ -1,11 +1,34 @@
 from typing import Optional
 from uuid import UUID
+from service.data_access_service.document_service import DocumentDataAccessService
 from registration.models import Document, DocumentType, Operation
 from service.data_access_service.operation_service import OperationDataAccessService
+from registration.utils import (
+    files_have_same_hash,
+)
+from django.core.files.base import ContentFile
 
 
 class DocumentService:
     @classmethod
     def get_existing_statutory_declaration_by_operation_id(cls, operation_id: UUID) -> Optional[Document]:
         operation: Operation = OperationDataAccessService.get_by_id(operation_id)
         return operation.documents.filter(type=DocumentType.objects.get(name="signed_statutory_declaration")).first()
+
+    @classmethod
+    def create_or_replace_operation_document(
+        cls, user_guid: UUID, operation_id: UUID, file_data: ContentFile, document_type: str
+    ) -> Document:
+        """This function receives a document and operation id. Operations only have one of each type of document, so this function uses the type to check if an existing document needs to be replaced, or if no document exists and one must be created. This function does NOT set any m2m relationships."""
+        existing_document = DocumentDataAccessService.get_operation_document_by_type(operation_id, document_type)
+        # if there is an existing  document, check if the new one is different
+        if existing_document:
+            # We need to check if the file has changed, if it has, we need to delete the old one and create a new one
+            if not files_have_same_hash(file_data, existing_document.file):
+                # brianna this could have been an update instead of a delete/create?
+                existing_document.delete()
```

Comment:
This could have been an update instead of a delete/recreate--possible tech debt

Reply:
sorry, I don't understand your comment...?

Reply:
A document update is being handled by deleting the existing document and then creating a new one. For other field updates, we  use the .update() method (or our custom one) and that sets the updated_at etc. columns. So this code is working fine, but it's not a true "update."